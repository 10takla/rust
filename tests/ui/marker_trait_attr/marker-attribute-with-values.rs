#![feature(marker_trait_attr)]

#[marker(always)] //~ ERROR malformed `marker` attribute
trait Marker1 {}

#[marker("never")] //~ ERROR malformed `marker` attribute
trait Marker2 {}

#[marker(key = "value")] //~ ERROR malformed `marker` attribute
trait Marker3 {}

#[marker]
trait Marker4 {}

#[marker(with_items)]
trait Marker5 {}

fn main() {}
