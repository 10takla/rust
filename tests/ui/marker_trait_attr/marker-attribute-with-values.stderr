error: malformed `marker` attribute input
  --> $DIR/marker-attribute-with-values.rs:3:1
   |
LL | #[marker(always)]
   | ^^^^^^^^^^^^^^^^^
   |
help: the following are the possible correct uses
   |
LL | #[marker(with_items)]
   | ~~~~~~~~~~~~~~~~~~~~~
LL | #[marker]
   | ~~~~~~~~~

error: malformed `marker` attribute input
  --> $DIR/marker-attribute-with-values.rs:6:1
   |
LL | #[marker("never")]
   | ^^^^^^^^^^^^^^^^^^
   |
help: the following are the possible correct uses
   |
LL | #[marker(with_items)]
   | ~~~~~~~~~~~~~~~~~~~~~
LL | #[marker]
   | ~~~~~~~~~

error: malformed `marker` attribute input
  --> $DIR/marker-attribute-with-values.rs:9:1
   |
LL | #[marker(key = "value")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: the following are the possible correct uses
   |
LL | #[marker(with_items)]
   | ~~~~~~~~~~~~~~~~~~~~~
LL | #[marker]
   | ~~~~~~~~~

error: aborting due to 3 previous errors

